### ISSUE TITLE
REST tutorial again: Can't resolve dependencies for zfcampus/statuslib-example

### ISSUE THREAD

**Issue by Rillke** [zfcampus/apigility-documentation#83]
When following the [REST Service tutorial](https://apigility.org/documentation/intro/first-rest-service), 
 `zfcampus/statuslib-example` requires `zendframework/zend-config ^2.6`, while `zendframework/zend-config` is somehow locked at 3.2.0.
```
someMac:Git user$ composer create-project zfcampus/zf-apigility-skeleton rbp-api2
Installing zfcampus/zf-apigility-skeleton (1.5.0)
  - Installing zfcampus/zf-apigility-skeleton (1.5.0): Loading from cache
Created project in rbp-api2
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 90 installs, 0 updates, 0 removals
  - Installing zendframework/zend-component-installer (2.1.1): Loading from cache
  - Installing zfcampus/zf-asset-manager (1.2.0): Loading from cache
  - Installing bshaffer/oauth2-server-php (v1.10.0): Loading from cache
  - Installing psr/link (1.0.0): Loading from cache
  - Installing zendframework/zend-stdlib (3.2.0): Loading from cache
  - Installing zendframework/zend-authentication (2.6.0): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing zendframework/zend-config (3.2.0): Loading from cache
  - Installing paragonie/random_compat (v2.0.12): Loading from cache
  - Installing zendframework/zend-math (3.1.0): Loading from cache
  - Installing container-interop/container-interop (1.2.0): Loading from cache
  - Installing zendframework/zend-crypt (3.3.0): Loading from cache
  - Installing zendframework/zend-db (2.9.3): Loading from cache
  - Installing zendframework/zend-escaper (2.6.0): Loading from cache
  - Installing zendframework/zend-json (3.1.0): Loading from cache
  - Installing zendframework/zend-loader (2.6.0): Loading from cache
  - Installing zendframework/zend-paginator (2.8.1): Loading from cache
  - Installing zendframework/zend-permissions-acl (2.7.0): Loading from cache
  - Installing zendframework/zend-permissions-rbac (3.0.0): Loading from cache
  - Installing zendframework/zend-servicemanager (3.3.2): Loading from cache
  - Installing zendframework/zend-validator (2.10.2): Loading from cache
  - Installing zendframework/zend-uri (2.6.1): Loading from cache
  - Installing zendframework/zend-http (2.8.0): Loading from cache
  - Installing zendframework/zend-router (3.0.2): Loading from cache
  - Installing zendframework/zend-eventmanager (3.2.1): Loading from cache
  - Installing zendframework/zend-view (2.10.0): Loading from cache
  - Installing zendframework/zend-modulemanager (2.8.2): Loading from cache
  - Installing zendframework/zend-mvc (3.1.1): Loading from cache
  - Installing zfcampus/zf-api-problem (1.3.0): Loading from cache
  - Installing zendframework/zend-filter (2.8.0): Loading from cache
  - Installing zfcampus/zf-content-negotiation (1.4.0): Loading from cache
  - Installing zfcampus/zf-configuration (1.3.3): Loading from cache
  - Installing zfcampus/zf-apigility-provider (1.3.0): Loading from cache
  - Installing zendframework/zend-inputfilter (2.8.1): Loading from cache
  - Installing michelf/php-markdown (1.8.0): Loading from cache
  - Installing zfcampus/zf-apigility-documentation (1.3.0): Loading from cache
  - Installing zfcampus/zf-development-mode (3.2.0): Loading from cache
  - Installing seld/jsonlint (1.7.1): Loading from cache
  - Installing justinrainbow/json-schema (1.6.1): Loading from cache
  - Installing herrera-io/json (1.0.3): Loading from cache
  - Installing kherge/version (1.0.1): Loading from cache
  - Installing myclabs/deep-copy (1.7.0): Loading from cache
  - Installing phar-io/version (1.0.1): Loading from cache
  - Installing phar-io/manifest (1.0.1): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
  - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing sebastian/exporter (3.1.0): Loading from cache
  - Installing sebastian/diff (3.0.0): Loading from cache
  - Installing sebastian/comparator (3.0.0): Loading from cache
  - Installing webmozart/assert (1.3.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.3.0): Loading from cache
  - Installing doctrine/instantiator (1.1.0): Loading from cache
  - Installing phpspec/prophecy (1.7.6): Loading from cache
  - Installing theseer/tokenizer (1.1.0): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/environment (3.1.0): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (3.0.0): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/php-file-iterator (1.4.5): Loading from cache
  - Installing phpunit/php-code-coverage (6.0.4): Loading from cache
  - Installing phpunit/php-timer (2.0.0): Loading from cache
  - Installing phpunit/phpunit-mock-objects (6.1.1): Loading from cache
  - Installing sebastian/global-state (2.0.0): Loading from cache
  - Installing sebastian/object-reflector (1.1.1): Loading from cache
  - Installing sebastian/object-enumerator (3.0.3): Loading from cache
  - Installing sebastian/resource-operations (1.0.0): Loading from cache
  - Installing zendframework/zend-debug (2.6.0): Loading from cache
  - Installing zendframework/zend-developer-tools (1.2.1): Loading from cache
  - Installing zendframework/zend-dom (2.7.1): Loading from cache
  - Installing zendframework/zend-console (2.7.0): Loading from cache
  - Installing phpunit/phpunit (7.1.5): Loading from cache
  - Installing zendframework/zend-test (3.2.0): Loading from cache
  - Installing zfcampus/zf-versioning (1.3.0): Loading from cache
  - Installing zfcampus/zf-rpc (1.4.0): Loading from cache
  - Installing zfcampus/zf-oauth2 (1.5.0): Loading from cache
  - Installing zfcampus/zf-mvc-auth (1.5.0): Loading from cache
  - Installing zendframework/zend-hydrator (2.4.0): Loading from cache
  - Installing zfcampus/zf-hal (1.5.0): Loading from cache
  - Installing zfcampus/zf-rest (1.4.0): Loading from cache
  - Installing zfcampus/zf-content-validation (1.4.0): Loading from cache
  - Installing zfcampus/zf-apigility-admin-ui (1.3.9): Loading from cache
  - Installing zfcampus/zf-apigility (1.4.0): Loading from cache
  - Installing zendframework/zend-code (3.3.0): Loading from cache
  - Installing zfcampus/zf-apigility-admin (1.6.0): Loading from cache
  - Installing zfcampus/zf-composer-autoloading (2.1.0): Loading from cache
  - Installing zfcampus/zf-console (1.4.0): Loading from cache
  - Installing herrera-io/phar-update (1.0.3): Loading from cache
  - Installing zfcampus/zf-deploy (1.3.0): Loading from cache
zfcampus/zf-apigility-skeleton suggests installing zfcampus/zf-apigility-doctrine (zfcampus/zf-apigility-doctrine ^2.2 to create Doctrine-Connected REST services)
zfcampus/zf-apigility-skeleton suggests installing zfcampus/zf-http-cache (zfcampus/zf-http-cache ^1.4 to add HTTP caching to your API)
zfcampus/zf-apigility-skeleton suggests installing zfr/zfr-cors (zfr/zfr-cors ^1.5 to add CORS support to your API)
bshaffer/oauth2-server-php suggests installing aws/aws-sdk-php (~2.8 is required to use DynamoDB storage)
bshaffer/oauth2-server-php suggests installing firebase/php-jwt (~2.2 is required to use JWT features)
bshaffer/oauth2-server-php suggests installing mongodb/mongodb (^1.1 is required to use MongoDB storage)
bshaffer/oauth2-server-php suggests installing predis/predis (Required to use Redis storage)
bshaffer/oauth2-server-php suggests installing thobbs/phpcassa (Required to use Cassandra storage)
zendframework/zend-authentication suggests installing zendframework/zend-ldap (Zend\Ldap component)
zendframework/zend-authentication suggests installing zendframework/zend-session (Zend\Session component)
zendframework/zend-config suggests installing zendframework/zend-i18n (^2.7.4; install if you want to use the Translator processor)
paragonie/random_compat suggests installing ext-libsodium (Provides a modern crypto API that can be used to generate random bytes.)
zendframework/zend-math suggests installing ext-gmp (If using the gmp functionality)
zendframework/zend-json suggests installing zendframework/zend-json-server (For implementing JSON-RPC servers)
zendframework/zend-json suggests installing zendframework/zend-xml2json (For converting XML documents to JSON)
zendframework/zend-paginator suggests installing zendframework/zend-cache (Zend\Cache component to support cache features)
zendframework/zend-servicemanager suggests installing ocramius/proxy-manager (ProxyManager 1.* to handle lazy initialization of services)
zendframework/zend-validator suggests installing zendframework/zend-i18n (Zend\I18n component to allow translation of validation error messages)
zendframework/zend-validator suggests installing zendframework/zend-i18n-resources (Translations of validator messages)
zendframework/zend-validator suggests installing zendframework/zend-session (Zend\Session component, ^2.8; required by the Csrf validator)
zendframework/zend-http suggests installing paragonie/certainty (For automated management of cacert.pem)
zendframework/zend-router suggests installing zendframework/zend-i18n (^2.6, if defining translatable HTTP path segments)
zendframework/zend-view suggests installing zendframework/zend-feed (Zend\Feed component)
zendframework/zend-view suggests installing zendframework/zend-i18n (Zend\I18n component)
zendframework/zend-view suggests installing zendframework/zend-navigation (Zend\Navigation component)
zendframework/zend-mvc suggests installing http-interop/http-middleware (^0.4.1 to be used together with zend-stratigility)
zendframework/zend-mvc suggests installing zendframework/zend-log (^2.9.1  To provide log functionality via LogFilterManager, LogFormatterManager, and LogProcessorManager)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-console (zend-mvc-console provides the ability to expose zend-mvc as a console application)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-i18n (zend-mvc-i18n provides integration with zend-i18n, including a translation bridge and translatable route segments)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-plugin-fileprg (To provide Post/Redirect/Get functionality around forms that container file uploads)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-plugin-flashmessenger (To provide flash messaging capabilities between requests)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-plugin-identity (To access the authenticated identity (per zend-authentication) in controllers)
zendframework/zend-mvc suggests installing zendframework/zend-mvc-plugin-prg (To provide Post/Redirect/Get functionality within controllers)
zendframework/zend-mvc suggests installing zendframework/zend-psr7bridge ((^0.2) To consume PSR-7 middleware within the MVC workflow)
zendframework/zend-mvc suggests installing zendframework/zend-servicemanager-di (zend-servicemanager-di provides utilities for integrating zend-di and zend-servicemanager in your zend-mvc application)
zendframework/zend-mvc suggests installing zendframework/zend-stratigility (zend-stratigility is required to use middleware pipes in the MiddlewareListener)
zendframework/zend-filter suggests installing zendframework/zend-i18n (Zend\I18n component for filters depending on i18n functionality)
phpunit/php-code-coverage suggests installing ext-xdebug (^2.6.0)
sebastian/global-state suggests installing ext-uopz (*)
zendframework/zend-debug suggests installing ext-xdebug (XDebug, for better backtrace output)
zendframework/zend-developer-tools suggests installing aist/aist-git-tools (Show you informations about current GIT repository)
zendframework/zend-developer-tools suggests installing bjyoungblood/bjy-profiler (Version: dev-master, allows the usage of the (Zend) Db collector.)
zendframework/zend-developer-tools suggests installing doctrine/doctrine-orm-module (Profile DoctrineORM queries)
zendframework/zend-developer-tools suggests installing jhuet/zdt-logger-module (Show you log data from Zend\Log)
zendframework/zend-developer-tools suggests installing ocramius/ocra-service-manager (OcraServiceManager can help you track dependencies within your application.)
zendframework/zend-developer-tools suggests installing san/san-session-toolbar (SanSessionToolbar can help you see current Zend\Session data you're using within your application.)
zendframework/zend-developer-tools suggests installing snapshotpl/zf-snap-event-debugger (ZfSnapEventDebugger can help you debug events from Zend\EventManager)
phpunit/phpunit suggests installing ext-xdebug (*)
phpunit/phpunit suggests installing phpunit/php-invoker (^2.0)
zendframework/zend-test suggests installing zendframework/zend-mvc-console (^1.1.8, to test MVC <-> console integration)
zfcampus/zf-oauth2 suggests installing alcaeus/mongo-php-adapter (^1.0.5, if you are using ext/mongodb and wish to use the MongoAdapter for OAuth2 credential storage.)
zendframework/zend-hydrator suggests installing zendframework/zend-serializer (^2.6.1, to use the SerializableStrategy)
zfcampus/zf-apigility-admin-ui suggests installing rwoverdijk/assetmanager (^1.7, to expose assets under the document root (may not yet be released))
zfcampus/zf-apigility suggests installing alcaeus/mongo-php-adapter (^1.0.5, if you are using ext/mongodb and wish to use the MongoConnectedListener.)
zfcampus/zf-apigility suggests installing rwoverdijk/assetmanager (^1.7, to install the assets used by zf-apigility-welcome and/or zf-apigility-admin-ui (may not yet be released))
zendframework/zend-code suggests installing doctrine/annotations (Doctrine\Common\Annotations >=1.0 for annotation features)
Package herrera-io/json is abandoned, you should avoid using it. Use kherge/json instead.
Package kherge/version is abandoned, you should avoid using it. No replacement was suggested.
Package herrera-io/phar-update is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
> zf-development-mode enable
You are now in development mode.
someMac:Git user$ cd rbp-api2
someMac:rbp-api2 user$ composer development-enable
> zf-development-mode enable
Already in development mode!
someMac:rbp-api2 user$ composer require zfcampus/statuslib-example
Using version ^1.1 for zfcampus/statuslib-example
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Can only install one of: zendframework/zend-config[2.6.0, 3.2.0].
    - Can only install one of: zendframework/zend-config[3.2.0, 2.6.0].
    - Can only install one of: zendframework/zend-config[2.6.0, 3.2.0].
    - zfcampus/statuslib-example 1.1.0 requires zendframework/zend-config ^2.6 -> satisfiable by zendframework/zend-config[2.6.0].
    - Installation request for zfcampus/statuslib-example ^1.1 -> satisfiable by zfcampus/statuslib-example[1.1.0].
    - Installation request for zendframework/zend-config (locked at 3.2.0) -> satisfiable by zendframework/zend-config[3.2.0].


Installation failed, reverting ./composer.json to its original content.
someMac:rbp-api2 user$ 
```

**Comment by Rillke** [comment]
Closing in favour of zfcampus/statuslib-example/issues/11
